misk/src/main/kotlin/misk/flags/DynamicFlagModule.kt

```diff
@@ -0,0 +1,163 @@
+package misk.flags
+
+import com.google.inject.TypeLiteral
+import com.google.inject.name.Names
+import com.squareup.moshi.Moshi
+import misk.inject.KAbstractModule
+import misk.inject.parameterizedType
+import javax.inject.Inject
+import javax.inject.Provider
+import kotlin.reflect.KClass
+import kotlin.reflect.KFunction
+
+/**
+ * Installs support for dynamic flag. Applications should inherit from this module and
+ * override [configureFlags], calling the various bindXXXFlags methods to make flag
+ * variables available
+ */
+abstract class DynamicFlagModule : KAbstractModule() {
+    override fun configure() {
+        requireBinding(DynamicFlagStore::class.java)
+        configureFlags()
+    }
+
+    protected abstract fun configureFlags()
+
+    inline fun <reified T : Flags> bindFlags(prefix: String = "", qualifier: Annotation? = null) =
+            bindFlags(T::class, prefix, qualifier)
+
+    fun <T : Flags> bindFlags(
+            kclass: KClass<T>,
+            prefix: String = "",
```

Comment:
should this have a default?

Reply:
I think we want to allow naked (unannotated) flag classes for e.g. application-level flags